### PR TITLE
Highlight urls in launch console

### PR DESCRIPTION
**Pull request by briandealwis**
Adds a URL link converter to the launch console to convert URLs into clickable links.

Goes part way to fixing #768.

### CODE REVIEW

File: plugins/com.google.cloud.tools.eclipse.appengine.localserver/plugin.xml [hunk 1]
```diff
@@ -145,6 +145,18 @@
       </consolePageParticipant>
    </extension>
    <extension
+         point="org.eclipse.ui.console.consolePatternMatchListeners">
+    <!-- pretty liberal URL pattern matcher -->
+    <consolePatternMatchListener
+      class="com.google.cloud.tools.eclipse.ui.util.console.PatternToHyperlinkConverter"
+      id="com.google.cloud.tools.eclipse.appengine.deploy.consolePatternMatchListener"
+      regex="https?://([\w-]+\.?)+(:\w+)?(/[^\s]*)?">
```
Comment (elharo) [inline plugins/com.google.cloud.tools.eclipse.appengine.localserver/plugin.xml]: I'm imagining a test that would extract the regex from the XML and compare it to strings.

**Comment by branflake2267** [comment]
Does the local dev server run with port https by default?

**Comment by briandealwis** [comment]
No, I don't think it supports https at all.  I kept it there just in case something logs a message with https.

**Comment by branflake2267** [comment]
Ah, gotcha. I was wondering, b/c we would have a mixed content issue if that were the case.

**Comment by briandealwis** [comment]
Hmm, I wonder if we could pull out the module/service information from the server logs to provide with [`org.eclipse.wst.server.core.model.IURLProvider`](https://github.com/gwt-plugins/gwt-eclipse-plugin/blob/master/plugins/com.gwtplugins.gwt.eclipse.wtp/src/com/google/gwt/eclipse/wtp/facet/GwtLaunchableAdapterDelegate.java#L34):
```
INFO     2016-07-28 01:33:21,421 dispatcher.py:197] Starting module "default" running at: http://localhost:8080
INFO     2016-07-28 01:33:21,424 admin_server.py:116] Starting admin server at: http://localhost:8000
```
[…]
```
Jul 28, 2016 1:33:23 AM com.google.appengine.tools.development.AbstractModule startup
INFO: Module instance default is running at http://localhost:54220/
Jul 28, 2016 1:33:23 AM com.google.appengine.tools.development.AbstractModule startup
INFO: The admin console is running at http://localhost:54220/_ah/admin
```
Unfortunately the log formats are different, and the `default` module port is different too.

**Pull request merged**